lib/scenic/command_recorder/statement_arguments.rb

```diff
@@ -36,15 +36,26 @@ def options
         @options ||= @args[1] || {}
       end
 
+      if Hash.respond_to? :ruby2_keywords_hash
+        def keyword_hash(hash)
+          Hash.ruby2_keywords_hash hash
```

Comment:
This should only be used if there is no other way essentially:
```
------------------------------------------------------------------------
  Hash.ruby2_keywords_hash(hash) -> hash

------------------------------------------------------------------------

Duplicates a given hash and adds a ruby2_keywords flag. This method is
not for casual use; debugging, researching, and some truly necessary
cases like deserialization of arguments.
```

Could whatever consumes the `options` from this class use them with `**options` when calling a method with it?
I think that should be sufficient.

OTOH, I can see preserving the ruby2_keywords state of the Hash also makes sense.

Reply:
> Could whatever consumes the options from this class use them with **options when calling a method with it?
I think that should be sufficient.

Not sure I understand. ActiveRecord's CommandRecorder ultimately passes them back in, which is why it keeps looking like ActiveRecord is broken. They get passed in to the methods in `Scenic::Statements`, mixed into... IDK, the connection or something, eg here: https://github.com/scenic-views/scenic/blob/v1.5.4/lib/scenic/statements.rb#L25

If you're asking if those methods could take `**options`, then yeah, if you're asking whether the caller (ActiveRecord) can change, then that's less likely, and it's not clear to me that's the right thing to do (b/c right now, ActiveRecord is just taking what you give it and giving it right back)... well... here, it gives it back to you, and asks you to give it the inverse invocation, which is what this code is doing.

I mean, I suppose you could mutate the original hash, as long as you know no one is expecting it to still represent the original invocation. But that feels super iffy.

Reply:
Part of the problem is that this keywordness isn't copied in cloning/duping:

```ruby
ruby2_keywords def a(*args)
  Hash.ruby2_keywords_hash? args[-1]        # => true
  Hash.ruby2_keywords_hash? args[-1].clone  # => false
  Hash.ruby2_keywords_hash? args[-1].dup    # => false
end
a b: 1
```

Reply:
It's hard for me to tell without investigating more. What would be useful for me to understand it better is to see the caller and callee code of where the flag gets lost (can be found out with `Hash.ruby2_keyword_hash?(h)`), and to see the callers of the `options` method from this class, and what it does with it.